### PR TITLE
Add new UT in PDisk for TEvChunkWrite contract

### PR DESCRIPTION
**Pull request by va-kuznecov**
### Changelog entry 

Fix PDisk didn't check TEvChunkWrite's offset and wrote data to with an unexpected offset. Add UT for that case.

### Changelog category 

* Bugfix 

### Additional information

...

### CODE REVIEW

File: ydb/core/blobstorage/pdisk/blobstorage_pdisk_ut.cpp [hunk 1]
```diff
@@ -1028,6 +1022,115 @@ Y_UNIT_TEST_SUITE(TPDiskTest) {
             Cerr << "all log writes are received" << Endl;
         }
     }
+
+    NPDisk::TEvChunkWrite::TPartsPtr GenParts(TReallyFastRng32& rng, size_t size) {
+        static int testCase = 0;
+        switch(testCase++) {
+            case 0: {
+                TBuffer x;
```
Comment (robdrynkin) [inline ydb/core/blobstorage/pdisk/blobstorage_pdisk_ut.cpp:1030]: not needed

File: ydb/core/blobstorage/pdisk/blobstorage_pdisk_ut.cpp [hunk 2]
```diff
@@ -1028,6 +1022,115 @@ Y_UNIT_TEST_SUITE(TPDiskTest) {
...
+                return MakeIntrusive<NPDisk::TEvChunkWrite::TBufBackedUpParts>(std::move(buffer));
+            }
+            case 1: {
+                size_t partsCount = rng.Uniform(1, 10);
+                TRope rope;
+                size_t createdBytes = 0;
+                if (size >= partsCount) {
+                    for (size_t i = 0; i < partsCount - 1; ++i) {
+                        TRope x(PrepareData(rng.Uniform(1, size / partsCount)));
+                        Cerr << x.size();
```
Comment (SammyVimes) [inline ydb/core/blobstorage/pdisk/blobstorage_pdisk_ut.cpp:1046]: Leftovers

**Review by SammyVimes — CHANGES_REQUESTED** [review]
also TStrokaBackedUpParts is still used somewhere

File: ydb/core/blobstorage/pdisk/blobstorage_pdisk_ut.cpp (hunk 1 above)
Comment (va-kuznecov) [reply ydb/core/blobstorage/pdisk/blobstorage_pdisk_ut.cpp:1030]: fixed

**Review by SammyVimes — APPROVED** [review]

**Pull request merged**